server/utils/utils.ts

```diff
@@ -193,6 +193,23 @@ const selectReleaseDate = (nomisRecord: Prisoner) => {
   return dateString
 }
 
+const selectReleaseDateFromLicence = (licence: Licence) => {
+  let dateString = licence.actualReleaseDate || licence.conditionalReleaseDate
+
+  if (!dateString) {
+    logger.error(`No release date found for NOMIS ID: ${licence.nomsId}`)
+    return 'not found'
+  }
+
+  try {
+    dateString = format(new Date(dateString), 'dd MMM yyyy')
+  } catch (e) {
```

Comment:
not sure about this error handling -could be wrong but seems unnecessarily defensive considering the licence date from CVL will always be `dd/MM/yyyy`? 

https://github.com/ministryofjustice/create-and-vary-a-licence-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/Licence.kt#L80

Reply:
I believe it's there in case the date comes back in a malformed state from NOMIS or elsewhere. We have specs to  test for that scenario, so my assumption is that it's something that does occasionally happen or has happened in the past.

Reply:
I had written a comment about this being in case the date comes back malformed from NOMIS, but in hindsight I think it's actually leftover from me copying the method from the selectReleaseDate method in the same file.
It may be overly defensive, but I'm also not sure it does any harm to leave it in place, to make sure the date in the CVL db isn't also malformed somehow?

Reply:
Unnecessarily defensive code just leads to more and more defensive code. 

Code like this is a statement that the format of dates from a licence is not known - which makes everywhere that doesn't express similar uncertainty look like a possible bug

The dates in the CVL DB are typed at `DateTime` and the API has consistent (if annoying) formatting rules so the uncertainty is unnecessary.

Reply:
If we have consistent formatting enforced in the API, then I'm happy to remove the uncertainty! I only kept it as there appears to be uncertainty in the data received from NOMIS and I wasn't aware that we enforced structure in the API.